src/pages/settings/Payments/PaymentMethodList.js

```diff
@@ -87,47 +87,66 @@ const defaultProps = {
     listHeaderComponent: null,
 };
 
-class PaymentMethodList extends Component {
-    constructor(props) {
-        super(props);
+/**
+ * Dismisses the error on the payment method
+ * @param {Object} item
+ */
+function dismissError(item) {
+    const paymentList = item.accountType === CONST.PAYMENT_METHODS.BANK_ACCOUNT ? ONYXKEYS.BANK_ACCOUNT_LIST : ONYXKEYS.CARD_LIST;
+    const paymentID = item.accountType === CONST.PAYMENT_METHODS.BANK_ACCOUNT ? lodashGet(item, ['accountData', 'bankAccountID'], '') : lodashGet(item, ['accountData', 'fundID'], '');
+
+    if (!paymentID) {
+        Log.info('Unable to clear payment method error: ', item);
+        return;
+    }
 
-        this.renderItem = this.renderItem.bind(this);
+    if (item.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
+        PaymentMethods.clearDeletePaymentMethodError(paymentList, paymentID);
+    } else {
+        PaymentMethods.clearAddPaymentMethodError(paymentList, paymentID);
     }
+}
 
-    /**
-     * @param {Boolean} isDefault
-     * @returns {*}
-     */
-    getDefaultBadgeText(isDefault = false) {
-        if (!isDefault) {
-            return null;
-        }
+/**
+ * @param {Array} filteredPaymentMethods
+ * @param {Boolean} isDefault
+ * @returns {Boolean}
+ */
+function shouldShowDefaultBadge(filteredPaymentMethods, isDefault = false) {
+    if (!isDefault) {
+        return false;
+    }
 
-        const defaultablePaymentMethodCount = _.reduce(
-            this.getFilteredPaymentMethods(),
-            (count, method) => (method.accountType === CONST.PAYMENT_METHODS.BANK_ACCOUNT || method.accountType === CONST.PAYMENT_METHODS.DEBIT_CARD ? count + 1 : count),
-            0,
-        );
-        if (defaultablePaymentMethodCount <= 1) {
-            return null;
-        }
+    const defaultablePaymentMethodCount = _.filter(
+        filteredPaymentMethods,
+        (method) => method.accountType === CONST.PAYMENT_METHODS.BANK_ACCOUNT || method.accountType === CONST.PAYMENT_METHODS.DEBIT_CARD,
+    ).length;
+    return defaultablePaymentMethodCount > 1;
+}
 
-        return this.props.translate('paymentMethodList.defaultPaymentMethod');
-    }
+/**
+ * @param {String} actionPaymentMethodType
+ * @param {String|Number} activePaymentMethodID
+ * @param {String} paymentMethod
+ * @return {Boolean}
+ */
+function isPaymentMethodActive(actionPaymentMethodType, activePaymentMethodID, paymentMethod) {
+    return paymentMethod.accountType === actionPaymentMethodType && paymentMethod.methodID === activePaymentMethodID;
+}
+function PaymentMethodList(props) {
+    const {actionPaymentMethodType, activePaymentMethodID, bankAccountList, cardList, filterType, network, onPress, payPalMeData, shouldShowSelectedState, selectedMethodID, translate} =
+        props;
```

Comment:
Deconstructing props to avoid this type of warning:

<img width="1063" alt="Screenshot 2023-06-05 at 12 54 12 PM" src="https://github.com/Expensify/App/assets/87341702/1a9d6f92-82d9-45a2-a4a9-0022a03100f1">